drafts/rxjs-memory-leaks.md

```diff
@@ -0,0 +1,223 @@
+---
+title: RxJS Memory Leaks in Angular
+domain: software-engineering-corner.hashnode.dev
+tags: web-development, rxjs, form, bugs-and-errors
+cover: https://cdn.hashnode.com/res/hashnode/image/stock/unsplash/xqFCy9AbHP4/upload/a6072d56d7c313baa3d0eafda3ebbd24.jpeg?w=1600&h=840&fit=crop&crop=entropy&auto=compress,format&format=webp
+publishAs: tispBe
+hideFromHashnodeCommunity: false
+ignorePost: true
+---
+
+Memory leaks do not sound like something you want in your Angular application. Yet, nearly all applications using [RxJS Observables](https://rxjs.dev/guide/observable) face the challenge of mitigating this threat. You might already wonder, what these memory leaks are and how they can affect your application. And what is more, how can you prevent them?
+
+This article describes RxJS Observable memory leaks and presents various techniques and patterns to tackle them in Angular applications.
+
+### Memory Leaks in a Nutshell
+
+In Angular web applications, memory leaks are caused by a mismanagement of [RxJs Observables](https://rxjs.dev/guide/observable). Thereby, allocated heap memory space is not freed up by the application when a component with an Observable is destroyed. This memory leak will result in performance issues, i.e. the page gets slower and slower. Higher load times negatively impact the user experience and increase the bounce rate [[1]](https://www.nngroup.com/articles/response-times-3-important-limits/). If a page load goes from 1s to 5s, then the bounce rate increases by 90% [[2]](https://www.thinkwithgoogle.com/consumer-insights/consumer-trends/mobile-page-speed-new-industry-benchmarks/). A page refresh will clear the heap memory again, and "reset" the memory leak. So, the longer an Angular component lives on your page, the higher the chance of experiencing a memory leak.
```

Comment:
Maybe phrase it slightly less absolute. Not every memory leak is an RxJs problem. Maybe use wording like "a big cause of memory leaks" or "are often caused by" or something similar.